frontend/src/views/c-zoom.vue

```diff
@@ -404,15 +410,16 @@ export default defineComponent({
     },
 
     updateFileTypes() {
+      // Assert that this.filteredUser is not undefined since this method is called in the created hook
       const commitsFileTypes = new Set<string>();
-      this.filteredUser.commits.forEach((commit) => {
+      this.filteredUser!.commits.forEach((commit) => {
         commit.commitResults.forEach((slice) => {
           Object.keys(slice.fileTypesAndContributionMap).forEach((fileType) => {
             commitsFileTypes.add(fileType);
           });
         });
       });
-      this.fileTypes = Object.keys(this.filteredUser.fileTypeContribution).filter(
+      this.fileTypes = Object.keys(this.filteredUser!.fileTypeContribution).filter(
```

Comment:
Since the non-null assertion operator is ignored during runtime, this method will throw an error if it is actually called when `this.filteredUser` is undefined.

It is true that in the current logic flow of our code, `this.filteredUser` will not be undefined (since you return early in `created()`), however there is no guarantee that this behaviour will be maintained in the future (e.g. If a developer mistakenly calls `updateFileTypes` before the return statement in `created`, or in another part of the code before `created` is run).

Hence it might be better (more type safety) to handle the undefined case explicitly instead - let me know what you think about this!

Related eslint rule that explains why using `!` might not be super ideal: https://typescript-eslint.io/rules/no-non-null-assertion/

Reply:
thanks for the insight! it seems then the best way to handle the case when `this.filteredUser == null` is to simply return early - what do you think?

Reply:
Yup, sounds good!

Reply:
ok, fixed!